### PR TITLE
[NAE-1868] Process role refresh

### PR DESCRIPTION
**Pull request by Retoocs**
# Description

Fixed aggregating permissions into map. Fixed situation, in which could user continue working on some task even if the admin removed his roles. Now he doesn't need to relog.

Fixes [NAE-1868]

## Dependencies

No new dependencies were introduced.

### Third party dependencies

No new dependencies were introduced.

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

Manually tested and partly by new unit tests

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |  Linux Mint 21.1 Cinnamon |
| Runtime             |  Java 11  |
| Dependency Manager  |   Maven 3.8.1  |
| Framework version   |    Spring Boot 2.7.8  |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @renczesstefan ...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [ ] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [ ] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1868]: https://netgrif.atlassian.net/browse/NAE-1868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/netgrif/application/engine/security/service/ISecurityContextService.java [hunk 1]
```diff
@@ -6,7 +6,7 @@ public interface ISecurityContextService {
 
     void saveToken(String token);
 
-    void reloadSecurityContext(LoggedUser loggedUser);
+    void reloadSecurityContext(LoggedUser loggedUser, boolean forceRefresh);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/security/service/ISecurityContextService.java:9]: For better readability you should define two public methods that will hide the boolean param:
```
void reloadSecurityContext(LoggedUser loggedUser)
void forceReloadSecurityContext(LoggedUser loggedUser)
```
Then in the implementation you can have a private method with boolean param:
```
public void reloadSecurityContext(LoggedUser loggedUser) {
   reloadSecurityContext(loggedUser, false);
}
public void forceReloadSecurityContext(LoggedUser loggedUser) {
   reloadSecurityContext(loggedUser, true);
}
private void reloadSecurityContext(LoggedUser loggedUser, boolean forceRefresh) {
   ...
}
```

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
     name: Test
     needs: build
     runs-on: ubuntu-latest
-    timeout-minutes: 90
+    timeout-minutes: 200
```
Comment (mazarijuraj) [inline .github/workflows/pr-build.yml:35]: Not sure if changes in this file should be merged, please check with devops

**Review by mazarijuraj — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-build.yml (hunk 2 above)
Comment (machacjozef) [reply .github/workflows/pr-build.yml:35]: these changes were made by me

its okey...

File: src/main/java/com/netgrif/application/engine/security/service/ISecurityContextService.java (hunk 1 above)
Comment (Retoocs) [reply src/main/java/com/netgrif/application/engine/security/service/ISecurityContextService.java]: Updated the code

**Review by mazarijuraj — APPROVED** [review]

**Pull request merged**